src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/webdriver/WebDriverUtils.kt

```diff
@@ -0,0 +1,48 @@
+package com.atlassian.performance.tools.jiraactions.api.webdriver
+
+import org.openqa.selenium.JavascriptExecutor
+import org.openqa.selenium.NoSuchElementException
+import org.openqa.selenium.WebDriver
+import org.openqa.selenium.WebElement
+import org.openqa.selenium.support.ui.ExpectedCondition
+import java.util.concurrent.atomic.AtomicInteger
+
+class WebDriverUtils {
```

Comment:
Utils is a dangerous name to use, because it doesn't have a responsibility and attracts random code.
I usually warn about the future evolution of such classes, but we can already see it here:
-  a bit of syntax sugar around `WebDriver`
- `ExpectedCondition` wrappers, which don't interact with `WebDriver` instances at all

I'd move the syntax sugar to `WebDriverExtension` (a sin I committed a long time ago 😉).
I'd extract the "sometimes" to `ExtraExpectedConditions`/`LightweightConditions` or even promote each to a named implementation.

PS. I thought I posted this comment hours ago. Seems I didn't send it. Sorry for the feedback delay.

Reply:
Also, since it's API, I gotta ask about the `companion object`. Do you aim to provide Java `static`? In order to mimic `org.openqa.selenium.support.ui.ExpectedConditions` static factory methods? If so, we gotta use [`@JvmStatic`](https://kotlinlang.org/docs/reference/java-to-kotlin-interop.html#static-methods).
And if we don't care about the class name - we can use [package-level functions](https://kotlinlang.org/docs/reference/java-to-kotlin-interop.html#package-level-functions) without an encompassing class, trading `LightweightConditions.Companion.sometimesApply` for `LightweightConditionsKt.sometimesApply` in Java.

Reply:
Added

Reply:
> Utils is a dangerous name to use, because it doesn't have a responsibility and attracts random code.
> I usually warn about the future evolution of such classes, but we can already see it here:
> 
> * a bit of syntax sugar around `WebDriver`
> * `ExpectedCondition` wrappers, which don't interact with `WebDriver` instances at all
> 
Actually, all these interact with webdriver (which is the reason for the performance problem behind them).

> I'd move the syntax sugar to `WebDriverExtension` (a sin I committed a long time ago 😉).

Yeah, I saw that class. I don't really like that approach, I think it should only be applied to methods which very closely resemble the methods already available on that class. For example:
`
fun WebDriver.maybeFindElement(
    locator: By
): WebElement? {
    return this.findElements(locator).singleOrNull()
}
`
Would be OK, but the methods in this utils class wouldn't.

> I'd extract the "sometimes" to `ExtraExpectedConditions`/`LightweightConditions` or even promote each to a named implementation.

OK.

Reply:
> Actually, all these interact with webdriver (which is the reason for the performance problem behind them).

Oops, you're right, I missed this bit: `fun apply(input: WebDriver?)`.
OTOH it's still a good idea to find a more specific home and focus for the code. Otherwise most classes here (and in `virtual-users` and `jira-software-actions`) would be merged into a multi-KLOC `WebDriverUtils` god-object 😉